test/mlir/onnx/onnx_lowering.mlir

```diff
@@ -2330,3 +2330,69 @@ func @test_resize1(%arg0 : tensor<3x4xf32>) -> tensor<*xf32> {
+// CHECK-DAG:       [[MEAN:%.+]] = arith.constant 0.000000e+00 : f64
+// CHECK-DAG:       [[SCALE:%.+]] = arith.constant 1.000000e+00 : f64
+// CHECK-DAG:       [[SEED:%.+]] = arith.constant 2.000000e+00 : f64
+// CHECK-DAG:       "krnl.random_normal"([[ALLOC]], [[ALL_VALUES]], [[MEAN]], [[SCALE]], [[SEED]]) : (memref<3x4x5xf64>, index, f64, f64, f64) -> ()
+// CHECK:           return [[ALLOC]] : memref<3x4x5xf64>
+}
+
+//-----
+  func @test_random_normal_like3(%arg0: tensor<3x4x5xf32>) -> tensor<*xf32> {
+    %0 = "onnx.RandomNormalLike"(%arg0) {dtype = 2 : si64, mean = 0.0 :f32, scale = 1.0 : f32} : (tensor<3x4x5xf32>) -> tensor<*xf32>
+    "std.return"(%0) : (tensor<*xf32>) -> ()
+// CHECK-LABEL:  @test_random_normal_like3
+// CHECK-DAG:       [[ALLOC:%.+]] = memref.alloc() {alignment = 16 : i64} : memref<3x4x5xf64>
+// CHECK-DAG:       [[ALL_VALUES:%.+]] = arith.constant 60 : index
+// CHECK-DAG:       [[MEAN:%.+]] = arith.constant 0.000000e+00 : f64
+// CHECK-DAG:       [[SCALE:%.+]] = arith.constant 1.000000e+00 : f64
+// CHECK-DAG:       [[SEED:%.+]] = arith.constant
+// CHECK-DAG:       "krnl.random_normal"([[ALLOC]], [[ALL_VALUES]], [[MEAN]], [[SCALE]], [[SEED]]) : (memref<3x4x5xf64>, index, f64, f64, f64) -> ()
+// CHECK:           return [[ALLOC]] : memref<3x4x5xf64>
+}
+
+//-----
+  func @test_random_normal_like4(%arg0: tensor<3x4x?x?xf32>) -> tensor<*xf32> {
+    %0 = "onnx.RandomNormalLike"(%arg0) {dtype = 1 : si64, mean = 0.0 :f32, scale = 1.0 : f32, seed = 2.0 : f32} : (tensor<3x4x?x?xf32>) -> tensor<*xf32>
+    "std.return"(%0) : (tensor<*xf32>) -> ()
+// CHECK-LABEL:  @test_random_normal_like4
+// CHECK-DAG:       [[C2:%.+]] = arith.constant 2 : index
+// CHECK-DAG:       [[DIM2:%.+]] = memref.dim %arg0, [[C2]] : memref<3x4x?x?xf32>
+// CHECK-DAG:       [[C3:%.+]] = arith.constant 3 : index
+// CHECK-DAG:       [[DIM3:%.+]] = memref.dim %arg0, [[C3]] : memref<3x4x?x?xf32>
+// CHECK-DAG:       [[DYN_ALLOC:%.+]] = memref.alloc([[DIM2]], [[DIM3]]) {alignment = 16 : i64} : memref<3x4x?x?xf32>
+// CHECK-DAG:       [[C12:%.+]] = arith.constant 12 : index
+// CHECK-DAG:       [[C2:%.+]] = arith.constant 2 : index
+// CHECK-DAG:       [[DIM2:%.+]] = memref.dim %arg0, [[C2]] : memref<3x4x?x?xf32>
+// CHECK-DAG:       [[MUL1:%.+]] = arith.muli [[C12]], [[DIM2]] : index
+// CHECK-DAG:       [[C3:%.+]] = arith.constant 3 : index
+// CHECK-DAG:       [[DIM3:%.+]] = memref.dim %arg0, [[C3]] : memref<3x4x?x?xf32>
+// CHECK-DAG:       [[ALL_VALUES:%.+]] = arith.muli [[MUL1]], [[DIM3]] : index
+// CHECK-DAG:       [[MEAN:%.+]] = arith.constant 0.000000e+00 : f32
+// CHECK-DAG:       [[SCALE:%.+]] = arith.constant 1.000000e+00 : f32
+// CHECK-DAG:       [[SEED:%.+]] = arith.constant 2.000000e+00 : f32
+// CHECK-DAG:       "krnl.random_normal"([[DYN_ALLOC]], [[ALL_VALUES]], [[MEAN]], [[SCALE]], [[SEED]]) : (memref<3x4x?x?xf32>, index, f32, f32, f32) -> ()
+// CHECK:           return [[DYN_ALLOC]] : memref<3x4x?x?xf32>
+}
```

Comment:
Can you pls add a test (or change one of the test cases) where `dtype` is not specified (The ONNX docs mention it is optional).